### DISCUSSION TITLE
Problems with Prior Constrains

### DISCUSSION THREAD

**Discussion by jgzga** [pymc-labs/pymc-marketing#1001] [General]
I'm attempting to add a constraint to the coefficient "intercept" variable, but it seems that it's not respecting the upper limit. Could I restrict the value of the coefficients with some kind of Prior?
```
my_model_config = {
    "intercept": Prior("HalfNormal", sigma=0.5).constrain(lower=0, upper=0.9),
    "gamma_fourier": Prior("Laplace", mu=0, b=2, dims="fourier_mode"),
    "beta_channel": Prior("HalfNormal", sigma=2, dims="channel"),
    "beta_control": Prior("Normal", mu=0, sigma=2, dims="control"),
    "likelihood": {
        "dist": "StudentT",
        "kwargs": {
            "sigma": Prior("HalfNormal", sigma=0.5),
            "nu": Prior("Gamma", alpha=25, beta=2)
        }
    }
}

```
This is the contribution graph, and as you can see, it's greater than .9, which is what I set in my constraint.
![descarga](https://github.com/user-attachments/assets/1921ccff-f005-45d9-9cf2-d6daae7c7c70)

Any help would be appreciated. Thanks

**Comment by williambdean** [comment]
Hi @jgzga 
Sorry the documentation seems to be confusing. The domain is not restricted but 95% of the density is put between lower and upper arguments. 
If you want the domain to change, use a different distribution because HalfNormal will allow for values >0

To see what you are currently using, try `Prior("HalfNormal", sigma=0.5).constrain(lower=0, upper=0.9).preliz.plot_pdf()` to plot

How would you like to put a prior on your intercept term? Maybe I can help formulate something

**Reply by jgzga** [reply]
I would like the domain of the intercept term to not have values higher than 0.9, but I still don't know how to do it, maybe by trying another distribution. But I'm not very short about it.
